contracts/lib/QuoteV3Auth/V3Parser.sol

```diff
@@ -154,4 +154,120 @@ library V3Parser {
 
         success = true;
     }
+
+    /// for structured quote parse
+    function validateParsedInput(V3Struct.ParsedV3QuoteStruct calldata v3Quote)
+        internal
+        pure
+        returns (
+            bool success,
+            V3Struct.Header memory header,
+            V3Struct.EnclaveReport memory localEnclaveReport,
+            bytes memory signedQuoteData, // concatenation of header and local enclave report bytes
+            V3Struct.ParsedECDSAQuoteV3AuthData memory authDataV3
+        )
+    {
+        success = true;
+        localEnclaveReport = v3Quote.localEnclaveReport;
+        V3Struct.EnclaveReport memory pckSignedQeReport = v3Quote.v3AuthData.pckSignedQeReport;
+
+        require(
+            localEnclaveReport.reserved3.length == 96 && localEnclaveReport.reserved4.length == 60
+                && localEnclaveReport.reportData.length == 64,
+            "local QE report has wrong length"
+        );
+        require(
+            pckSignedQeReport.reserved3.length == 96 && pckSignedQeReport.reserved4.length == 60
+                && pckSignedQeReport.reportData.length == 64,
+            "QE report has wrong length"
+        );
+        require(
+            v3Quote.v3AuthData.certification.certType == 5,
+            "certType must be 5: Concatenated PCK Cert Chain (PEM formatted)"
+        );
+        require(v3Quote.v3AuthData.certification.decodedCertDataArray.length == 3, "3 certs in chain");
+        require(
+            v3Quote.v3AuthData.ecdsa256BitSignature.length == 64 && v3Quote.v3AuthData.ecdsaAttestationKey.length == 64
+                && v3Quote.v3AuthData.qeReportSignature.length == 64,
+            "Invalid ECDSA signature format"
+        );
+        require(
+            v3Quote.v3AuthData.qeAuthData.parsedDataSize == v3Quote.v3AuthData.qeAuthData.data.length,
+            "Invalid QEAuthData size"
+        );
+
+        // todo!
```

Comment:
TODO: We should look into the gas cost to perform the base64 encoding on the individual certificates here for the length check.

Reply:
I think we can make this as a separate functionality to work on, since the scope of this PR is strictly on parsing the quote.